### PR TITLE
#2420. Exhaustiveness variable tests. Add switch statement check

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Extension-types/exhaustiveness_variable_A01_t05.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+extension type AET1(A _) {}
+extension type AET2(A _) implements A {}
+extension type BET1(B _) {}
+extension type BET2(B _) implements B {}
+
+String test1(AET1 a) {
+  switch (a) {
+    case B _: return 'B';
+  }
+}
```
Comment (eernstg) [inline LanguageFeatures/Extension-types/exhaustiveness_variable_A01_t05.dart:33]: This one is blocked on https://github.com/dart-lang/language/issues/3534: It hasn't yet been decided whether we should use the extension type erasures (as I'd prefer) during flow analysis in order to determine whether or not it is possible to reach the end of the function body in this example.

So we'll have to put this one in a waiting position until https://github.com/dart-lang/language/issues/3534 has been settled.

**Review by eernstg — COMMENTED** [review]
Blocked on https://github.com/dart-lang/language/issues/3534.

Perhaps split this into one PR that uses return type `String` on a bunch of existing test functions containing switch expressions, and put the new tests into a different PR? We could then land the return-type part, and confine the blockedness to the second PR.

File: LanguageFeatures/Extension-types/exhaustiveness_variable_A01_t05.dart (hunk 1 above)
Comment (sgrekhov) [reply LanguageFeatures/Extension-types/exhaustiveness_variable_A01_t05.dart:33]: https://github.com/dart-lang/language/issues/3534 has been settled. Now this test should work

**Review by sgrekhov — COMMENTED** [review]
I believe now this PR is nublocked. Please take another look

File: LanguageFeatures/Extension-types/exhaustiveness_variable_A01_t05.dart (hunk 1 above)
Comment (eernstg) [reply LanguageFeatures/Extension-types/exhaustiveness_variable_A01_t05.dart:33]: Right!

**Review by eernstg — APPROVED** [review]
LGTM!

**Pull request merged**